examples/java-matter-controller/java/src/com/matter/controller/commands/common/MatterCommand.java

```diff
@@ -0,0 +1,112 @@
+/*
+ *   Copyright (c) 2022 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+package com.matter.controller.commands.common;
+
+import com.matter.controller.config.PersistentStorage;
+import com.matter.controller.config.PersistentStorageOpCertStore;
+import com.matter.controller.config.PersistentStorageOperationalKeystore;
+import java.util.Optional;
+
+public abstract class MatterCommand extends Command {
+
+  public static final short kMaxGroupsPerFabric = 5;
+  public static final short kMaxGroupKeysPerFabric = 8;
+
+  protected PersistentStorage mDefaultStorage = new PersistentStorage();
```

Comment:
Can these be private and final, and then surfaced to subclasses via protected methods for encapsulation?

```
private PersistentStorage mDefaultStorage = new PersistentStorage();

protected PersistentStorage getPersistentStorage() { return mDefaultStorage; }
```

This allows you to make safer assertions about the lifetime and state of these properties.